knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -125,6 +131,18 @@ func (r *ReconcileKnativeKafka) reconcileKnativeKafka(instance *operatorv1alpha1
 	return nil
 }
 
+// set a finalizer to clean up cluster-scoped resources and resources from other namespaces
+func (r *ReconcileKnativeKafka) ensureFinalizers(instance *operatorv1alpha1.KnativeKafka) error {
```

Comment:
Now we're actually setting the finalizer on the CR.
This is necessary to be able to remove resources that can't be GC'ed because they're in a different namespace (`knative-sources` namespace)

Reply:
IMO it would be nicer, if we just stash them all to the `knative-eventing` namespace.

I will propose the same upstream - insted of the `knative-sources` namespace ,....

Reply:
well, we still need the finalizer, so, this code won't change.
There are also cluster scoped resources which can't be GC'ed and we still need the finalizer to prevent the deletion of KnativeKafka CR before them.

Happy to discuss the get rid of `knative-sources` to `knative-eventing` for KafkaSource.

IMO we should do that change separately once everything is working. It might require some upstream changes perhaps, I don't know yet.